ChaosMod/Effects/db/Vehs/VehsBusJesus.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+	Effect by ProfessorBiddle, a very shite programmer
+*/
+
+#include <stdafx.h>
+
+static void OnStart()
+{
+	Ped playerPed = PLAYER_PED_ID();
+
+
+	//delete current vehicle
+	if (IS_PED_IN_ANY_VEHICLE(playerPed, false))
+	{
+		Vehicle playerVeh = GET_VEHICLE_PED_IS_IN(playerPed, false);
+		SET_ENTITY_AS_MISSION_ENTITY(playerVeh, true, true);
+		DELETE_VEHICLE(&playerVeh);

```

Comment:
Deleting the current vehicle should be avoided, as it can remove a mission car and cause a softlock

Reply:
If I don't the bus gets stuck on top of the other vehicle, I am a shit programmer how do I fix this? :P

Reply:
That created some collisiony issues, so I botched it even further and _gently_ pushed the vehicle out of the way.

SET_ENTITY_COORDS_NO_OFFSET(playerVeh, oldPos.x, oldPos.y, oldPos.z + 5, 0, 0, 0);
		SET_ENTITY_VELOCITY(playerVeh, vel.x, vel.y, 40.f);